services/tone_analyzer/v3.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright 2013,2016 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+module.exports = function (RED) {
+  var watson = require('watson-developer-cloud');  
+  var cfenv = require('cfenv');
+
+  // Require the Cloud Foundry Module to pull credentials from bound service 
+  // If they are found then they are stored in sUsername and sPassword, as the 
+  // service credentials. This separation from sUsername and username to allow 
+  // the end user to modify the node credentials when the service is not bound.
+  // Otherwise, once set username would never get reset, resulting in a frustrated
+  // user who, when he errenously enters bad credentials, can't figure out why
+  // the edited ones are not being taken.
+
+  // Not ever used, and codeacy complains about it.
+
+  var username, password, sUsername, sPassword;
+
+  var service = cfenv.getAppEnv().getServiceCreds(/tone analyzer/i)
+
+  if (service) {
+    sUsername = service.username;
+    sPassword = service.password;
+  }
+
+  // Node RED Admin - fetch and set vcap services
+  RED.httpAdmin.get('/watson-tone-analyzer/vcap', function (req, res) {
+    res.json(service ? {bound_service: true} : null);
+  });
+
+  // Function that checks the payload and determines
+  // whether it is JSON or a Buffer
+  var checkPayload = function(payload) {
+    var message = null;
+
+    var hasJSONmethod = (typeof payload.toJSON === 'function') ;
+      if (hasJSONmethod === true) {
+        if (payload.toJSON().type === 'Buffer') {
+          isBuffer = true;
+        }      
+      }      
+      // Payload (text to be analysed) must be a string (content is either raw string or Buffer)
+      if (typeof payload !== 'string' &&  isBuffer !== true) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["isBuffer" is not defined.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2653030321/issues/source?bid=3225115&fileBranchId=3325196#l57)